testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPReadOnlyTest.java

```diff
@@ -125,6 +127,23 @@ public void testReadOnlyWithTOTPEnabled() {
         user.removeCredential(totpCredentialId);
     }
 
+    // KEYCLOAK-3365
+    @Test
+    public void testReadOnlyUserDoesNotThrowIfUnchanged() {
+        UserResource user = ApiUtil.findUserByUsernameId(testRealm(), "johnkeycloak");
```

Comment:
Do you please have a chance to update this test a bit and test that requiredAction add/remove works? As I believe that is the main reason why you created this PR. I mean something similar to this:
```
        UserResource user = ApiUtil.findUserByUsernameId(testRealm(), "johnkeycloak");
        UserRepresentation userRepresentation = user.toRepresentation();
        user.setRequiredActions(Collections.singletonList(UserModel.RequiredAction.CONFIGURE_TOTP.toString());
        user.update(userRepresentation);

        // Check and revert required action
        userRepresentation = user.toRepresentation();
        Assert.assertEquals(user.toRepresentation().getRequiredActions().getSize(), 1);
        Assert.assertEquals(user.toRepresentation().getRequiredActions().get(0), UserModel.RequiredAction.CONFIGURE_TOTP.toString());
        user.setRequiredActions(Collections.emptyList());
        user.update(userRepresentation);
```

Reply:
Good point! Done.